onnxruntime/core/graph/graph.cc

```diff
@@ -5537,6 +5537,42 @@ Graph::Graph(const Model& owning_model,
       is_loaded_from_model_file_(true) {  // true as the Graph isn't manually constructed from scratch
 }
 
+size_t Graph::ComputeNodeMemoryUsage(NodeIndex node_idx) const {
+  /// XXX: In some cases some kernels can copy its attributes to a device
+  // those are edge cases which we currently do not account for.
+  const Node* node = GetNode(node_idx);
+  if (node != nullptr) {
+    SafeInt<size_t> result = 0;
+    for (const auto* input : node->InputDefs()) {
+      if (input->Exists()) {
+        // Let's see if this is an initializer
+        constexpr const bool check_outer_scope_true = true;
+        const ONNX_NAMESPACE::TensorProto* initializer =
+            GetConstantInitializer(input->Name(), check_outer_scope_true);
+        if (initializer != nullptr) {
+          size_t out;
+          if (utils::GetSizeInBytesFromTensorProto<0>(*initializer, &out).IsOK()) {
```

Comment:
If an initializer is used by two nodes, will the size of initializer be counted twice?